### PR TITLE
Added Models and Controllers

### PR DESCRIPTION
**Pull request by ZachCarrillo**
Added Customer, CustomerGroup, and GroupMember models and controllers for each

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -348,3 +348,4 @@ MigrationBackup/
 
 # Ionide (cross platform F# VS Code tools) working folder
 .ionide/
+CustomerContactBook/CustomerContactBook/obj/project.assets.json
```
Comment (ysemerikov) [inline .gitignore:351]: Feel free to copy `.gitignore` from [my repository](https://github.com/ysemerikov/oa-customer-contact-book/blob/main/.gitignore)

File: CustomerContactBook/CustomerContactBook/Controllers/CustomerGroupsController.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            {
+                if (!CustomerGroupExists(id))
+                {
+                    return NotFound();
+                }
+                else
+                {
+                    throw;
+                }
+            }
```
Comment (ysemerikov) [inline CustomerContactBook/CustomerContactBook/Controllers/CustomerGroupsController.cs]:
```suggestion
            catch (DbUpdateConcurrencyException) when (!CustomerGroupExists(id))
            {
                return NotFound();
            }
```

File: CustomerContactBook/CustomerContactBook/Controllers/CustomerGroupsController.cs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+            _context.Groups.Remove(customerGroup);
+            await _context.SaveChangesAsync();
+
+            return NoContent();
+        }
+
+        private bool CustomerGroupExists(int id)
+        {
+            return _context.Groups.Any(e => e.Id == id);
```
Comment (ysemerikov) [inline CustomerContactBook/CustomerContactBook/Controllers/CustomerGroupsController.cs:97]: I see in most places you use `Async` version of methods to access database, why not here?

File: CustomerContactBook/CustomerContactBook/Models/Customer.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        public int Id { get; set; }
+        public string? FirstName { get; set; }
+        public string? LastName { get; set; }
+        public string FullName
+        {
+            get
+            {
+                return LastName + ", " + FirstName;
+            }
+        }
```
Comment (ysemerikov) [inline CustomerContactBook/CustomerContactBook/Models/Customer.cs]:
```suggestion
        public string FullName => LastName + ", " + FirstName;
```

File: CustomerContactBook/CustomerContactBook/Models/Customer.cs [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+        public string? PhoneNum { get; set; }
+        public string? Email { get; set; }
+        public string? Address { get; set; }
+        public DateTime BirthDay { get; set; }
```
Comment (ysemerikov) [inline CustomerContactBook/CustomerContactBook/Models/Customer.cs]:
```suggestion
        public DateTime? BirthDay { get; set; }
```

File: CustomerContactBook/CustomerContactBook/Models/Customer.cs [hunk 6]
```diff
@@ -0,0 +1,20 @@
+﻿namespace CustomerContactBook.Models
+{
+    public class Customer
+    {
+        public int Id { get; set; }
```
Comment (ysemerikov) [inline CustomerContactBook/CustomerContactBook/Models/Customer.cs]: `int.MaxValue` is small for database identifier, please use `long`

File: CustomerContactBook/CustomerContactBook/Controllers/CustomerGroupsController.cs [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+using Microsoft.EntityFrameworkCore;
+using CustomerContactBook.Models;
+
+namespace CustomerContactBook.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class CustomerGroupsController : ControllerBase
+    {
+        private readonly CustomerGroupContext _context;
```
Comment (ysemerikov) [inline CustomerContactBook/CustomerContactBook/Controllers/CustomerGroupsController.cs:16]: FYI, DatabaseContext shouldn't be used in the Controller level, there should be CustomerService.
Anyway it's fine for now, at the next steps we will allocate service layer.

File: CustomerContactBook/CustomerContactBook/Controllers/CustomerGroupsController.cs [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+                return NotFound();
+            }
+
+            return customerGroup;
+        }
+
+        // PUT: api/CustomerGroups/5
+        // To protect from overposting attacks, see https://go.microsoft.com/fwlink/?linkid=2123754
+        [HttpPut("{id}")]
+        public async Task<IActionResult> PutCustomerGroup(int id, CustomerGroup customerGroup)
```
Comment (ysemerikov) [inline CustomerContactBook/CustomerContactBook/Controllers/CustomerGroupsController.cs:47]: `CustomerGroup` is class that directly mapped into database table, for controllers please use another one (`CustomerGroupDto`), like described in [the link](https://go.microsoft.com/fwlink/?linkid=2123754) from comment of this method.

So there will be:
```
public class CustomerGroup // this class maps to database
{
    public long Id { get; set; }
    public string Name { get; set; }
}

public class CustomerGroupDto // this class is parameter of controller methods
{
    public long Id { get; set; }
    public string Name { get; set; }
}

public class CustomerGroupCreateDto // this class is parameter of controller methods
{
    public string Name { get; set; }
}
```

**Review by ysemerikov — COMMENTED** [review]
I glad you start to use EF to access database, I will tell more about this on the future sessions.

File: CustomerContactBook/CustomerContactBook/Controllers/CustomerGroupsController.cs (hunk 2 above)
Comment (ZachCarrillo) [reply CustomerContactBook/CustomerContactBook/Controllers/CustomerGroupsController.cs]: I have made this charge to each controller and pushed it to this branch

**Review by ysemerikov — APPROVED** [review]
synced in the slack:
'DTO' classes will be introduced in the next PR
everything else is fine to merge now.

**Pull request merged**